crypto/keys/hd/hdpath.go

```diff
@@ -55,13 +54,99 @@ func NewParams(purpose, coinType, account uint32, change bool, addressIdx uint32
 	}
 }
 
+// Parse the BIP44 path and unmarshal into the struct.
+// nolint: gocyclo
+func NewParamsFromPath(path string) (*BIP44Params, error) {
+	spl := strings.Split(path, "/")
+	if len(spl) != 5 {
+		return nil, fmt.Errorf("path length is wrong. Expected 5, got %d", len(spl))
+	}
+
+	if spl[0] != "44'" {
+		return nil, fmt.Errorf("first field in path must be 44', got %v", spl[0])
+	}
+
+	if !isHardened(spl[1]) || !isHardened(spl[2]) {
+		return nil,
+			fmt.Errorf("second and third field in path must be hardened (ie. contain the suffix ', got %v and %v", spl[1], spl[2])
+	}
+	if isHardened(spl[3]) || isHardened(spl[4]) {
+		return nil,
+			fmt.Errorf("fourth and fifth field in path must not be hardened (ie. not contain the suffix ', got %v and %v", spl[3], spl[4])
+	}
```

Comment:
all these if statements could be lumped into a switch statement - make it a bit more concise

Reply:
How's that?

Reply:
Having all those if statements lined up (which have returns, so they effectively act as chained if-else statements) act identically as a switch statement - I like compactness where it's appropriate but doesn't super matter here

Reply:
Can you show me what the switch statement would look like? Each of these conditions is on a different piece of the slice, so not clear to me what we could switch on